token/cli/src/main.rs

```diff
@@ -1712,127 +1666,78 @@ async fn command_gc(
                         .unwrap_or_else(|err| panic!("Invalid token amount: {}", err));
 
-                    let close_authority = ui_token_account.close_authority.map_or(owner, |s| {
-                        s.parse::<Pubkey>()
-                            .unwrap_or_else(|err| panic!("Invalid close authority: {}", err))
-                    });
-
                     let entry = accounts_by_token.entry(token).or_insert_with(HashMap::new);
                     entry.insert(
                         token_account,
-                        (
-                            token_amount,
-                            ui_token_account.token_amount.decimals,
-                            frozen,
-                            close_authority,
-                        ),
+                        (token_amount, ui_token_account.token_amount.decimals, frozen),
                     );
                 }
             }
         }
     }
 
-    let mut instructions = vec![];
-    let mut lamports_needed = 0;
+    let mut results = vec![];
+    for (token_pubkey, accounts) in accounts_by_token.into_iter() {
+        println_display(config, format!("Processing token: {}", token_pubkey));
 
-    for (token, accounts) in accounts_by_token.into_iter() {
-        println_display(config, format!("Processing token: {}", token));
-        let associated_token_account =
-            get_associated_token_address_with_program_id(&owner, &token, &config.program_id);
+        let token = token_client_from_config(config, &token_pubkey);
+        let associated_token_account = token.get_associated_token_address(&owner);
         let total_balance: u64 = accounts.values().map(|account| account.0).sum();
 
-        if total_balance > 0 && !accounts.contains_key(&associated_token_account) {
-            // Create the associated token account
-            instructions.push(vec![create_associated_token_account(
-                &config.fee_payer.pubkey(),
-                &owner,
-                &token,
-                &config.program_id,
-            )]);
-            lamports_needed += minimum_balance_for_rent_exemption;
-        }
+        for (address, (amount, decimals, frozen)) in accounts {
+            let is_associated = address == associated_token_account;
 
-        for (address, (amount, decimals, frozen, close_authority)) in accounts {
-            match (
-                address == associated_token_account,
-                close_empty_associated_accounts,
-                total_balance > 0,
-            ) {
-                (true, _, true) => continue, // don't ever close associated token account with amount
-                (true, false, _) => continue, // don't close associated token account if close_empty_associated_accounts isn't set
-                (true, true, false) => println_display(
-                    config,
-                    format!("Closing Account {}", associated_token_account),
-                ),
-                _ => {}
```

Comment:
this is painfully confusing, my series of if statements is more verbose but i think easier to read